farmdata2_modules/fd2_tabs/resources/RegexInputComponent.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * A Vue component for input HTML elements that use Regex to validate user input
+ * 
+ * <p><b>data-cy attributes</b></p>
+ * <table>
+ * <thead><tr><th>Value</th>        <th>Descripion</th></tr></thead>
+ * <tbody>
+ * <tr><td>dropdown-component</td>  <td>div containing the label and select elements.</td></tr>
+ * <tr><td>dropdown-input</td>      <td>The select element.</td></tr>
+ * <tr><td>optioni</td>             <td>The ith option element, i=0,1,2...</td></tr>
+ * </tbody>
+ * </table>
+ * 
+ * @vue-prop {RegExp} [regExp] - The regular expression used to validate user inputs.
+ * @vue-prop {String} [defaultVal] - Used to clear the page once a log has been submitted.
+ * @vue-prop {String} [setType] - Defines the type of input box.
+ * @vue-prop {String} [setColor] - Defines the background color of the input element.
+ * @vue-prop {String} [setHeight] - Defines the height of the input element.
+ * @vue-prop {String} [setWidth] - Defines the width of the input element.
+ * @vue-prop {String} [setMin] - Defines the min of the input element.
```

Comment:
Min, max, step props should indicate that they are ignored if not applicable to the type of the input element.